domain/application/service/exposed.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+
+	"github.com/juju/collections/set"
+
+	"github.com/juju/juju/domain/application"
+	internalerrors "github.com/juju/juju/internal/errors"
+)
+
+// ApplicationExposed returns wheter the provided application is exposed or not.
+//
+// If no application is found, an error satisfying
+// [applicationerrors.ApplicationNotFound] is returned.
+func (s *Service) ApplicationExposed(ctx context.Context, appName string) (bool, error) {
+	appID, err := s.st.GetApplicationIDByName(ctx, appName)
+	if err != nil {
+		return false, internalerrors.Capture(err)
+	}
+
+	return s.st.ApplicationExposed(ctx, appID)
+}
+
+// ExposedEndpoints returns map where keys are endpoint names (or the ""
+// value which represents all endpoints) and values are ExposedEndpoint
+// instances that specify which sources (spaces or CIDRs) can access the
+// opened ports for each endpoint once the application is exposed.
+//
+// If no application is found, an error satisfying
+// [applicationerrors.ApplicationNotFound] is returned.
+func (s *Service) ExposedEndpoints(ctx context.Context, appName string) (map[string]application.ExposedEndpoint, error) {
+	appID, err := s.st.GetApplicationIDByName(ctx, appName)
+	if err != nil {
+		return nil, internalerrors.Capture(err)
+	}
+
+	return s.st.ExposedEndpoints(ctx, appID)
+}
+
+// UnsetExposeSettings removes the expose settings for the provided list of
+// endpoint names. If the resulting exposed endpoints map for the application
+// becomes empty after the settings are removed, the application will be
+// automatically unexposed.
+//
+// If no application is found, an error satisfying
+// [applicationerrors.ApplicationNotFound] is returned.
+func (s *Service) UnsetExposeSettings(ctx context.Context, appName string, exposedEndpoints set.Strings) error {
+	appID, err := s.st.GetApplicationIDByName(ctx, appName)
+	if err != nil {
+		return internalerrors.Capture(err)
+	}
+
+	return s.st.UnsetExposeSettings(ctx, appID, exposedEndpoints)
+}
+
+// MergeExposeSettings marks the application as exposed and merges the provided
+// ExposedEndpoint details into the current set of expose settings. The merge
+// operation will overwrite expose settings for each existing endpoint name.
+//
+// If no application is found, an error satisfying
+// [applicationerrors.ApplicationNotFound] is returned.
+func (s *Service) MergeExposeSettings(ctx context.Context, appName string, exposedEndpoints map[string]application.ExposedEndpoint) error {
+	appID, err := s.st.GetApplicationIDByName(ctx, appName)
+	if err != nil {
+		return internalerrors.Capture(err)
+	}
+
+	return s.st.MergeExposeSettings(ctx, appID, exposedEndpoints)
+}
```

Comment:
Considering we don't support upgrading (refreshing) an application, when is this called?

Reply:
That's actually where we write in the db when we `expose`: https://github.com/juju/juju/blob/main/apiserver/facades/client/application/application.go#L1257

Do you think we should rename this?

Reply:
No this is fine.